src/java/com/palantir/cassandra/check/VolumeIntegrityCheck.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.cassandra.check;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.UUID;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.palantir.cassandra.utils.FileParser;
+import com.palantir.logsafe.Preconditions;
+import org.apache.cassandra.config.DatabaseDescriptor;
+
+/**
+ * Check if the backing volume has been tampered with. While Cassandra supports multiple data drive, we only
+ * use 1 data drive.
+ */
+public final class VolumeIntegrityCheck
+{
+    public static final String METADATA_NAME = "cassandra-metadata.json";
+
+    private final UUID hostId;
+
+    private final FileParser<VolumeMetadata> dataDriveMetadataFileParser;
+
+    private final FileParser<VolumeMetadata> commitLogMetadataFileParser;
+
+    static class VolumeMetadataType extends TypeReference<VolumeMetadata>
+    {
+    }
```

Comment:
Might be overkill, considering we seem to be only using this in one place

Reply:
I thought this would be more readable than trying to inline. What do you think?

Reply:
I am in favor of inlining when using one liners or one off logic. It makes the code more readable imo since you don't have to search for the type/function definition